demux/demux_mkv.c

```diff
@@ -1962,7 +1963,7 @@ static int demux_mkv_open_sub(demuxer_t *demuxer, mkv_track_t *track)
     sh->codec->extradata = track->private_data;
     sh->codec->extradata_size = track->private_size;
 
-    if (!strcmp(sh->codec->codec, "arib_caption") && track->private_size >= 3) {
+    if (subtitle_type && !strcmp(sh->codec->codec, "arib_caption") && track->private_size >= 3) {
```

Comment:
please don't put a crash fix into the same comment as a no-op change and then title the commit with that one

Reply:
Well the other line is not really a no-op. It prevents the crash on that file as well and changes the error message output so it's not spammed multiple times. But of course this line is needed anyways and is better. I can change the commit title to something like `demux_mkv: don't crash on unknown subtitle` instead?

Reply:
Updated.